src/server/validate.py

```diff
@@ -76,9 +78,24 @@ def validate_lang_and_year(lang, year):
 
     supported_langs = [lan.lang_code for lan in (SUPPORTED_LANGUAGES.get(year) or [DEFAULT_LANGUAGE])]
 
-    # If an unsupported language code is passed in, abort.
+    # If an unsupported language code is passed in, check if we have similar.
     if lang is not None and lang not in supported_langs:
         logging.debug('Unsupported language set: %s.' % lang)
+
+        # Check it's not just a simple case issue
+        if lang.lower() in supported_langs:
+            return (lang.lower(), year)
+
+        # Handle lookups for special cases (e.g. Chinese)
+        if lang.lower() in LANGUAGE_MAPPING:
+            return (LANGUAGE_MAPPING.get(lang.lower()), year)
```

Comment:
Shouldn't we add a check that the `LANGUAGE_MAPPING.get(lang.lower())` is in `supported_langs` just to avoid any possible errors.

Reply:
Good idea in case someone enters bad config. Added.